libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java

```diff
@@ -59,11 +67,23 @@ public static EntitlementChecker checker() {
     public static void initialize(Instrumentation inst) throws Exception {
         manager = initChecker();
 
-        Map<MethodKey, CheckMethod> checkMethods = INSTRUMENTER_FACTORY.lookupMethods(EntitlementChecker.class);
+        Map<MethodKey, CheckMethod> checkMethods = new HashMap<>(INSTRUMENTATION_SERVICE.lookupMethods(EntitlementChecker.class));
```

Comment:
very optional nit... Looks like `InstrumenterImpl` later contains the same mutable hash map. how about returning a custom interface from `lookupMethods` rather exposing the map and having to deal with im/mutability, e.g.

```
interface InstrumentedMethods {
  void add(InstrumentationInfo instrumentationInfo); // add dynamic instrumentation by mutating map internally
  Map<MethodKey, CheckMethod> asMap(); // immutable copy for use in InstrumenterImpl
}
```

Reply:
Interesting idea, I'll try it out